test/unit/control/UPausable.test.ts

```diff
@@ -0,0 +1,102 @@
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
+import { expect } from 'chai'
+import HRE from 'hardhat'
+
+import { MockUPausable, MockUPausable__factory } from '../../../types/generated'
+
+const { ethers } = HRE
+
+describe.only('UPausable', () => {
+  let owner: SignerWithAddress
+  let newOwner: SignerWithAddress
+  let user: SignerWithAddress
+  let uPausable: MockUPausable
+
+  beforeEach(async () => {
+    ;[owner, newOwner, user] = await ethers.getSigners()
+    uPausable = await new MockUPausable__factory(owner).deploy()
+  })
+
+  describe('#UPausable__initialize', async () => {
+    it('initializes pauser', async () => {
+      expect(await uPausable.pauser()).to.equal(ethers.constants.AddressZero)
+
+      await expect(uPausable.connect(owner).__initialize()).to.emit(uPausable, 'PauserUpdated').withArgs(owner.address)
+
+      expect(await uPausable.pauser()).to.equal(owner.address)
+      expect(await uPausable.owner()).to.equal(owner.address)
+    })
+  })
+
+  describe('#updatePauser', async () => {
+    beforeEach(async () => {
+      await uPausable.connect(owner).__initialize()
+    })
+
+    it('updates pauser', async () => {
+      await expect(uPausable.connect(owner).updatePauser(newOwner.address))
+        .to.emit(uPausable, 'PauserUpdated')
+        .withArgs(newOwner.address)
+
+      expect(await uPausable.pauser()).to.equal(newOwner.address)
+    })
+
+    it('only pauser can update pauser', async () => {
```

Comment:
nit: only owner

Reply:
maybe actually add another revert test here to verify the pauser also can't update the pauser